app/services/slack_service.rb

```diff
@@ -0,0 +1,200 @@
+module SlackService
+  class InvalidRequest < RuntimeError; end
+
+  class InvalidCommand < RuntimeError; end
+
+  public
+
+  def self.create_post(command_text, slack_team_id, slack_user_id)
```

Comment:
This method is still quite long. From a readers perspective all the error handling is not that important, so you could extract this into helper methods, for example like this:


```rb
team = find_team(slack_team_id)
sender = find_sender(slack_user_id)
receivers = get_receivers_from_command(command_text)
message = parse_message(command_text)
amount = parse_amount(command_text)

...
```

Thay way as I reader of this code I can quickly graps what this is doing and I can always drill down into one of these helper methods if I'm interested in how exactly the team is retrieved.

If you apply this consistently you'll end up with a code base that is much easier to grasp for the developers coming after you.